client/app/scripts/superdesk/editor/editor.js

```diff
@@ -550,16 +550,375 @@ function EditorService(spellcheck, $rootScope, $timeout) {
+                        // create a new block containing the embed
+                        return vm.editorCtrl.insertNewBlockAfter(vm.blockBefore, {
+                            blockType: 'embed',
+                            embedType: embedType,
+                            body: [
+                                '<figure>',
+                                embedBody,
+                                '</figure>'
+                            ].join('\n')
+                        });
+                    },
+                    createBlockFromEmbed: function() {
+                        vm.retrieveEmbed().then(function(embed) {
+                            vm.createFigureBlock(embed.provider, embed.body);
+                            // close the addEmbed form
+                            vm.toggle(true);
+                        });
+                    },
+                    createBlockFromPicture: function(picture) {
+                        vm.createFigureBlock('Image', [
+                            '<img alt="' + picture.description + '" src="' + picture.renditions.viewImage.href + '">',
+                            '<figcaption>' + picture.headline + '</figcaption>'
+                        ].join('\n'));
+                    }
+                });
+            }],
+            link: function(scope, element, attrs, controllers) {
+                angular.extend(controllers[0], {
+                    blockBefore: scope.block,
+                    editorCtrl: controllers[1]
+                });
+            }
+        };
+    })
+    .directive('sdTextEditorDropZone', function () {
+        return {
+            scope: true,
+            require: '^sdAddEmbed',
+            link: function(scope, element, attrs, ctrl) {
+                var PICTURE_TYPE = 'application/superdesk.item.picture';
+                element
+                .on('drop', function(event) {
+                    event.preventDefault();
+                    var item = angular.fromJson(event.originalEvent.dataTransfer.getData(PICTURE_TYPE));
+                    ctrl.createBlockFromPicture(item);
+                    element.removeClass('drag-active');
+                })
+                .on('dragover', function(event) {
+                    if (event.originalEvent.dataTransfer.types[0] === PICTURE_TYPE) {
+                        event.preventDefault();
+                        element.addClass('drag-active');
+                    }
+                })
+                .on('dragleave', function(event) {
+                    element.removeClass('drag-active');
+                });
+            }
+        };
+    })
+    .directive('sdTextEditor', ['$timeout', 'lodash', function ($timeout, _) {
+        return {
+            scope: {type: '=', config: '=', language: '='},
+            require: ['sdTextEditor', 'ngModel'],
+            templateUrl: 'scripts/superdesk/editor/views/editor.html',
+            controllerAs: 'vm',
```

Comment:
Since using `controllerAs`, consider combining this with `bindToController: true` so that the directive properties get bound to the controller instead of the scope.